### PR TITLE
Topic sk jcs 13808

### PR DESCRIPTION
**Pull request by skommala**
[JCS-13808] - Network validation script fails when using NSGs which have rules with other NSGs as source

Testing is still in progress which I will update in jira.

### CODE REVIEW

File: terraform/modules/network-validator/scripts/network_validation.sh [hunk 1]
```diff
@@ -250,14 +258,18 @@ function check_tcp_port_open_in_seclist_or_nsg() {
...
+        return
       fi
     done
   fi
-
-  echo 1
+  if [[ ${#nsg_sec_list_array[@]} != 0 ]]
+  then
+    echo "${nsg_sec_list_array[@]}"
+  else
```
Comment (HarshadVilekar) [inline terraform/modules/network-validator/scripts/network_validation.sh:270]: If the first ingress rule has NSG as a source - then the warning array is populated (echo warning).
If the second ingress rule has a CIDR block as a source, and the port is not open (not in the CIDR range) - then it's an error condition (echo 1).

However, won't this block only print a warning, and the error condition is missed?

Is this case tested?

Comment (telake) [reply terraform/modules/network-validator/scripts/network_validation.sh:270]: I agree that this is very ugly, but we need a quick fix. My comments on why I think this will "solve" this issue for now:
case 1 - nsg with port open to CIDR range. Error 0 returned.
case 2 - all nsg's use CIDR ranges, but none are the range we need. Error 1 returned.
case 3 - some nsg's reference other nsgs, and some reference CIDR ranges, but none that reference CIDR ranges are the range we need. Warning returned, because it is indeterminate unless we recursively search the CIDR references until we get to a CIDR. We will have a task in the backlog to do this in the future.

File: terraform/modules/network-validator/scripts/network_validation.sh [hunk 2]
```diff
@@ -1021,17 +1090,20 @@ then
     res=$(validate_subnet_port_access "${LB_SUBNET_2_OCID}" ${LB_PORT} "${LB_SOURCE_CIDR}")
     if [[ $res -ne 0 ]]
     then
-      echo "ERROR: Port [$LB_PORT] is not open for ${LB_SOURCE_CIDR} in LB Subnet CIDR [${LB_SUBNET_2_OCID}]. ${NETWORK_VALIDATION_MSG}"
-      validation_return_code=2
+      echo "WARNING: Port [$LB_PORT] is not open for ${LB_SOURCE_CIDR} in LB Subnet CIDR [${LB_SUBNET_2_OCID}]. ${NETWORK_VALIDATION_MSG}"
```
Comment (telake) [inline terraform/modules/network-validator/scripts/network_validation.sh:1093]: This seems unrelated to the change. Please make sure that this is the correct change required.

**Review by telake — APPROVED** [review]

**Review by HarshadVilekar — APPROVED** [review]
